src/sentry/tasks/embeddings_grouping/utils.py

```diff
@@ -168,17 +167,33 @@ def get_current_batch_groups_from_postgres(project, last_processed_group_index,
             total_groups_to_backfill_length,
         )
 
+    groups_to_backfill_not_deleted_status = [
+        (group_id, data)
+        for (group_id, data, status) in groups_to_backfill_batch
+        if status not in [GroupStatus.PENDING_DELETION, GroupStatus.DELETION_IN_PROGRESS]
+    ]
```

Comment:
this check could be combined with the following embedding check, but I kept it separate for logging purposes. let me know if you think we should combine them